### PR TITLE
Adding independent scrolling for the left nav and the main content area

### PR DESCRIPTION
**Pull request by sijieamoy**
## Changes
1. Allow the left nav and the main content areas to scroll separately
2. Reduce the size of the Ray logo to show more of the left nav TOC
3. Remove the "full screen mode" button that's unclear why it's useful
4. Fix some minor CSS formatting issues

cc @scottsun94 

Before: 
image

After:
image

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by pcmoritz** [comment]
@maxpumperla I think we should merge the PR, can you have a look? There is a conflict with one of your search changes, you probably know best what to do here :)

**Comment by maxpumperla** [comment]
@sijieamoy thanks for this. The one part I think we should think about is this one:

Screenshot 2022-09-15 at 08 59 19

we introduced a horizontal top bar that extends to the very right. It cuts off the scroll bar, and in my opinion somewhat disrupts the flow of the two remaining logos (more difficult to see the alignment of them).

File: doc/source/_static/css/custom.css [hunk 1]
```diff
@@ -1,57 +1,102 @@
...
-  overflow-y: visible;
-  height: 100% !important;
+  overflow-y: auto;
+  position: inherit;
+}
+#site-navigation.collapse {
+  margin-left: -186px;
+}
+.topbar {
+  z-index: 1900;
```
Comment (maxpumperla) [inline doc/source/_static/css/custom.css:9]: I think this causes my nit, what's the reasoning for having that? (not against it per se, just want to understand)

File: doc/source/conf.py [hunk 2]
```diff
@@ -195,6 +195,7 @@
     "use_repository_button": True,
     "use_issues_button": True,
     "use_edit_page_button": True,
+    "use_fullscreen_button": False,
```
Comment (maxpumperla) [inline doc/source/conf.py:202]: I actually sometimes use this to read longer tutorials distraction-free...

**Comment by maxpumperla** [comment]
@sijieamoy I also noticed that in the mobile view the search box pop-up is now "full screen". I like this, but I vaguely remember that people were complaining about this, which is why I made it not overflow into the main content area.

**Pull request closed without merging**